snow/consensus/snowball/tree.go

```diff
@@ -414,16 +414,23 @@ func (u *unaryNode) RecordPoll(votes bag.Bag[ids.ID], reset bool) (node, bool) {
 		u.shouldReset = true // Make sure my child is also reset correctly
 	}
 
-	if votes.Len() < u.tree.params.Alpha {
+	numVotes := votes.Len()
+	if numVotes < u.tree.params.AlphaPreference {
 		// I didn't get enough votes, I must reset and my child must reset as
 		// well
 		u.snowball.RecordUnsuccessfulPoll()
 		u.shouldReset = true
 		return u, false
 	}
 
-	// I got enough votes this time
-	u.snowball.RecordSuccessfulPoll()
+	if numVotes < u.tree.params.AlphaConfidence {
+		// I got enough votes to update my preference, but not increase my
+		// confidence.
+		u.snowball.RecordPollPreference()
+	} else {
+		// I got enough votes to increase my confidence
+		u.snowball.RecordSuccessfulPoll()
+	}
```

Comment:
nit: maybe a switch would be cleaner here? May be just my personal preference, but this are all conditions over numVotes and I'd like them to be more grouped. numVotes may be scoped in the switch statement only.

Reply:
nit: also maybe it's margially more readable to list cases as:

1. numVotes >= u.tree.params.AlphaConfidence
2. numVotes >=  u.tree.params.AlphaPreference
3. default